### PR TITLE
[IMP] mail: Added "New Message" command/hotkey

### PR DESCRIPTION
**Pull request by FrancoisGe**
Task-ID: 2597153

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/services/new_message_service/new_message_service.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { registry } from "@web/core/registry";
+
+const { Component } = owl;
+
+export const newMessageService = {
+    dependencies: ["command"],
+    start(newEnv, { command }) {
+        command.add("Message a user", () => {
+            const env = Component.env;
+            const messagingMenu = env.messaging && env.messaging.messagingMenu;
```
Comment (seb-odoo) [inline addons/mail/static/src/services/new_message_service/new_message_service.js:12]: Until we have proper new service dependency for messaging, this should also await on both messaging created and initialized.
Then you can remove the condition on env.messaging.

File: addons/mail/static/src/services/new_message_service/new_message_service.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            if (!env.messaging.device.isMobile) {
+                env.messaging.chatWindowManager.openNewMessage();
+                messagingMenu.close();
+            } else {
+                messagingMenu.toggleMobileNewMessage();
+            }
+
```
Comment (seb-odoo) [inline addons/mail/static/src/services/new_message_service/new_message_service.js:19]: no need for new line here

File: addons/mail/static/src/services/new_message_service/new_message_service.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        }, {
+            category: "mail",
+            hotkey: "alt+shift+w",
```
Comment (seb-odoo) [inline addons/mail/static/src/services/new_message_service/new_message_service.js:22]: Just a note but that shortcut is almost impossible to use in azerty, with 3 keys that are right next to each other on the corner of the keyboard. But I don't really care as I'll always be using ctrl+k anyway.

File: addons/mail/static/src/services/new_message_service/new_message_service.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+export const newMessageService = {
+    dependencies: ["command"],
+    start(newEnv, { command }) {
+        command.add("Message a user", () => {
+            const env = Component.env;
+            const messagingMenu = env.messaging && env.messaging.messagingMenu;
+            if (!env.messaging.device.isMobile) {
+                env.messaging.chatWindowManager.openNewMessage();
+                messagingMenu.close();
```
Comment (seb-odoo) [inline addons/mail/static/src/services/new_message_service/new_message_service.js:15]: I don't think it should specifically close the menu? It was the case when opening it from the menu itself by convenience, but this looks like an unrelated side effect in this case.

File: addons/mail/static/src/services/new_message_service/new_message_service.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            } else {
+                messagingMenu.toggleMobileNewMessage();
+            }
```
Comment (seb-odoo) [inline addons/mail/static/src/services/new_message_service/new_message_service.js:18]: Did you actually test this in mobile? The shortcut doesn't work at all, and opening it from ctrl+k only works if the messaging menu is already opened. Also it has the side effect of closing it if it's already open.
I think for mobile it should do both open menu + display the mobile search (and not with toggles).

**Review by seb-odoo — CHANGES_REQUESTED** [review]
Thank you for the PR, overall it looks fine in the basic case.

See below for complete test/validation.

Also it would be nice to have a basic test here.

File: addons/mail/static/src/services/new_message_service/new_message_service.js (hunk 5 above)
Comment (FrancoisGe) [reply addons/mail/static/src/services/new_message_service/new_message_service.js:18]: I just checked with Nathan, shortcuts are not accessible from mobile. I'll remove the logic related to this.

File: addons/mail/static/src/services/new_message_service/new_message_service.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import { registry } from "@web/core/registry";
+
+const { Component } = owl;
+
+export const newMessageService = {
+    dependencies: ["command"],
+    start(newEnv, { command }) {
+        command.add("Message a user", () => {
+            const env = Component.env;
+            env.messaging.chatWindowManager.openNewMessage();
```
Comment (seb-odoo) [inline addons/mail/static/src/services/new_message_service/new_message_service.js:12]: still need to add the 2 await as on the other PR, without them using shortcut before messaging is created would lead to a crash

File: addons/mail/static/src/services/new_message_service/new_message_service.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+
+const { Component } = owl;
+
+export const newMessageService = {
```
Comment (seb-odoo) [inline addons/mail/static/src/services/new_message_service/new_message_service.js:7]: Do we really need a new service for this actually? This should be part of ChatWindowService maybe while legacy, or even messaging service directly.

I don't see any usage of newEnv in the code, so having it in a legacy service would be ok?

File: addons/mail/static/src/services/new_message_service/new_message_service_tests.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
+/** @odoo-module **/
+
+import { newMessageService } from "./new_message_service";
```
Comment (seb-odoo) [inline addons/mail/static/src/services/new_message_service/new_message_service_tests.js:3]: I much prefer never using relative import, they are way too annoying to read (checking context becomes important), impossible to grep (to look where a file is potentially imported), and hard to maintain (moving files is a pain).

File: addons/mail/static/src/services/new_message_service/new_message_service.js (hunk 7 above)
Comment (FrancoisGe) [reply addons/mail/static/src/services/new_message_service/new_message_service.js:7]: we need access to the "command" service which is in the new env.

File: addons/mail/static/src/services/new_message_service/new_message_service.js (hunk 6 above)
Comment (FrancoisGe) [reply addons/mail/static/src/services/new_message_service/new_message_service.js:12]: I don't understand the use case ? openNewMessage isn't async ?

Comment (seb-odoo) [reply addons/mail/static/src/services/new_message_service/new_message_service.js:12]: I meant
```
        await this.env.messagingCreatedPromise;
        await this.env.messaging.initializedPromise;
```

Comment (seb-odoo) [reply addons/mail/static/src/services/new_message_service/new_message_service.js:12]: Calling `openNewMessage()` only works if discuss app is not open. If discuss app is open, we'll need to call a new equivalent method to the currently existing `onClickMobileNewChatButton`.

Comment (FrancoisGe) [reply addons/mail/static/src/services/new_message_service/new_message_service.js:12]: How to know if discuss app is open ?

Comment (FrancoisGe) [reply addons/mail/static/src/services/new_message_service/new_message_service.js]: If I want to do something like `onClickMobileNewChatButton`, I need to have access to discuss.How can I do this?

Comment (seb-odoo) [reply addons/mail/static/src/services/new_message_service/new_message_service.js]: `env.messaging.discuss.isOpen`

File: addons/mail/static/src/services/new_message_service/new_message_service.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    dependencies: ["command"],
+    start(newEnv, { command }) {
+        command.add("Message a user", async () => {
+            const env = Component.env;
+            if (env.messaging.discuss.isOpen){
+                env.messaging.discuss.update({ isAddingChat: true });
+            } else {
+                await env.messagingCreatedPromise;
+                await env.messaging.initializedPromise;
```
Comment (seb-odoo) [inline addons/mail/static/src/services/new_message_service/new_message_service.js:16]: This should be done outside of the condition.

Comment (seb-odoo) [reply addons/mail/static/src/services/new_message_service/new_message_service.js:16]: I'll just push the change and merge it now I guess.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+
